### ISSUE TITLE
can not open or new file

### ISSUE THREAD

**Issue by tgly307** [DUOLabs333/Photopea-Offline#2]
I clone this repo and run it in web server( nginx ), the chrome can open the page, but can not new file or open file locally

**Comment by DUOLabs333** [comment]
Huh, you're right. Don't know how I didn't catch that. I'll look into this. Thank you!

**Comment by coolcraftnet18** [comment]
If you're gonna get mad and make it offline, at least make the offline version work

**Comment by DUOLabs333** [comment]
@tgly307 Ok, I changed how I downloaded the website, so it should work now.

**Comment by coolcraftnet18** [comment]
What did you even change to make it work offline? Also, can you remove this banner 
![image](https://user-images.githubusercontent.com/18317131/106509285-6c9eff80-649b-11eb-9121-4022c888256d.png)

**Comment by coolcraftnet18** [comment]
Also, CORS is a problem, which means you may need to use localhost 
![image](https://user-images.githubusercontent.com/18317131/106509796-0f577e00-649c-11eb-9bac-976d6060e1fd.png)

**Comment by DUOLabs333** [comment]
@coolcraftnet18 Oh, I didn't change anything. I just downloaded it offline. And for the banner, you can just remove the .alertpanel in the all.css, but that would remove all alert panels. The cause of this panel is because DarcyRipper doesn't have an option to wait before downloading, so it saves the page before it gets removed. I could find the place where the Loading takes place and remove it, but the JS is minifiied and obfuscated, so it may take awhile (unless you can help me search for it :) )?

**Comment by coolcraftnet18** [comment]
Analytics.js is what is slowing the loading time

**Comment by coolcraftnet18** [comment]
You also need to remove the ad banner from the css

**Comment by DUOLabs333** [comment]
About the localhost, I used python's http.server, but now nothing works at all (though the "Loading" panel disappears). Also, the ad banner (I use ublock, so I don't see ads when I visit the actual site)?

**Comment by DUOLabs333** [comment]
Ah, I think I see it-- ad_unit, right?

**Comment by coolcraftnet18** [comment]
Probably, I don't know why a localhost server messes things up though 🤔,
it did that to me too

**Comment by DUOLabs333** [comment]
I don't see an analytics.js

**Comment by coolcraftnet18** [comment]
It might be pulling it from the web

**Comment by DUOLabs333** [comment]
i wonder what would happen then if i disconnect from the internet?

**Comment by coolcraftnet18** [comment]
No idea, you can disconnect quickly through network and sharing center

**Comment by DUOLabs333** [comment]
Huh, you were right-- if I disconnect, it loads much faster.

**Comment by DUOLabs333** [comment]
So it is pulling (or at least trying to pull) from the web.

**Comment by coolcraftnet18** [comment]
You should be able to chop that out

**Comment by DUOLabs333** [comment]
Ok, fixed that problem. Now I just need to find where the loading banner is made.

**Comment by coolcraftnet18** [comment]
Alright

**Comment by tgly307** [comment]
@DUOLabs333 thx for apply. Could you download this website? This one is more clean than photopea official website. https://ps.gaoding.com/sources/index.html I tried this one but cant open menu as before.

**Comment by tgly307** [comment]
And if you don't mind, could you share your download method?

**Comment by coolcraftnet18** [comment]
> @DUOLabs333 thx for reply. Could you download this [website](https://ps.gaoding.com/sources/index.html)? This one is more clean than photopea official website. I tried this one but cant open menu as before.

Do you speak chinese? DarcyRipper is the download method btw

**Comment by coolcraftnet18** [comment]
https://lelinhtinh.github.io/de4js/

**Comment by tgly307** [comment]
> > @DUOLabs333 thx for reply. Could you download this [website](https://ps.gaoding.com/sources/index.html)? This one is more clean than photopea official website. I tried this one but cant open menu as before.
> 
> Do you speak chinese? DarcyRipper is the download method btw

Yes, but I dont mind which language the tool use, I prefer this one is this is cleaner than the official one. Thanks for DarcyRipper, it is very useful.

**Comment by DUOLabs333** [comment]
@coolcraftnet18 Thanks, I'll check it out.

**Comment by tgly307** [comment]
I have a question, I use DarcyRipper to download the website, the folder is like this.
![image](https://user-images.githubusercontent.com/25153311/106549613-4f763980-654c-11eb-8386-0fd381b78ad8.png)
when I use the root folder as web root, the url is like `localhost:8080/ps.gaoding.com/`, it works.
but when I use ps.gaoding.com folder as web root, the url is like `localhost:8080`, it cant open menu as before.
What different between these? I tried to delete other two folder, it also works, why it must have the sub url?

**Comment by DUOLabs333** [comment]
@tgly307 Well, it makes sense. The web root has to have all the resources to load as usual. ps.goading.com doesn't have all the resources needed, but the root folder does.

**Comment by tgly307** [comment]
> @tgly307 Well, it makes sense. The web root has to have all the resources to load as usual. ps.goading.com doesn't have all the resources needed, but the root folder does.

But when I remove the two fonts folder, it works as well as before, so I guess the issue is not about resources. 
And I enter the `ps.gaoding.com` folder, rename the `sources` folder to `photopea.com`, and use `ps.gaoding.com` folder as web root, it works. I guess there is some code to determine if the path contains a specific keyword.

**Comment by DUOLabs333** [comment]
@tgly307 Maybe the index.html has paths to `photopea.com`, so it looks for files in that folder with that name? I don't know.

**Comment by coolcraftnet18** [comment]
Dunno, but with localhost it works offline perfectly

**Comment by DUOLabs333** [comment]
Really? I couldn't get mine to work with localhost.

**Comment by coolcraftnet18** [comment]
![image](https://user-images.githubusercontent.com/18317131/106617950-0b2f6d00-653d-11eb-8082-a6c215d6e363.png)

**Comment by coolcraftnet18** [comment]
I wonder if it is even legal to make an offline version of this software? I can easily wrap this into an application, but I question whether I should

**Comment by DUOLabs333** [comment]
Oh, it works! Don't know how it got fixed.

**Comment by DUOLabs333** [comment]
Ok, thank you all so much. I updated the README accordingly. If there is no other problems, I will close this issue in 2 hours/

**Comment by coolcraftnet18** [comment]
I still wonder how Ivan would make money if there is an offline version of this software

**Comment by DUOLabs333** [comment]
@tgly307 Ok, now I know what you meant. You're right, it only works when www.photopea.com is not the root folder. Unknown why.

**Comment by tgly307** [comment]
I tried to locate the location of the path determination statement, but failed. I suspect it is some kind of translation of the string. I have tried the Chinese version and the original version. The Chinese version can contain the `gaoding` string, but the original version cannot, which indicates that there is path judgment

**Comment by tgly307** [comment]
So far, it works perfectly offline, although the path doesn't look very elegant.

**Comment by DUOLabs333** [comment]
Yeah, got mine to work as well. Now, to work on getting fonts installed, as they are not downloaded by default.

**Comment by coolcraftnet18** [comment]
https://github.com/photopea/photopea/issues/45

**Comment by DUOLabs333** [comment]
So, I'm guessing distributing the source code is a no-go, right? What about if we package it up as a WARC file? Sure, it still may be possible to extract the pages, but it would be harder to.

**Comment by coolcraftnet18** [comment]
By downloading it, we are technically redistributing the source code? I think if we were to unobfuscate it, it would be even worse.

**Comment by DUOLabs333** [comment]
Ah, IANAL, so I'm not sure. I don't think just downloading it is redistribution. Downloading the source code is no more illegal than visiting the site, as we are only grabbing what our browsers can access.

**Comment by coolcraftnet18** [comment]
Well, you downloaded and uploaded it

**Comment by DUOLabs333** [comment]
From what I can find: "It's not illegal to download complete websites. How you're permitted to use that downloaded data is controlled by the copyright and terms & conditions posted on the website, if any. If no such notices are listed (highly unlikely), there there is no restriction on the use of the website's content." So, just downloading it is legal (probably because it is technically what our browsers do), but how we use it is up to Photopea. I will have to look into that.

**Comment by DUOLabs333** [comment]
There actually doesn't seem to be a Terms of Services though, which is highly interesting.

**Comment by coolcraftnet18** [comment]
Yeah, it would be kinda shitty if a ton of forks of photopea started
popping up, and then he wouldn't make money

**Comment by coolcraftnet18** [comment]
Click the account button

**Comment by DUOLabs333** [comment]
Ah, there it is. Still, there doesn't seem to be anything saying we can't. Ethically though, ehhhh...

**Comment by coolcraftnet18** [comment]
I know you have to have a license to put it on your website, if you click
account and scroll down you'll see more

**Comment by DUOLabs333** [comment]
Ah, the self-hosted package. Hmmm...

**Comment by DUOLabs333** [comment]
Closing the issue now.

**Comment by pluja** [comment]
I can't get this to work. I can open the page and load everything (menus, etc.) but I can't create a new file.
